client/my-sites/comments/comment-list/index.jsx

```diff
@@ -31,7 +38,57 @@ export class CommentList extends Component {
 		}
 	}
 
+	getNoticeContent = ( commentId, newStatus, previousStatus ) => {
+		const { translate } = this.props;
+		const options = {
+			duration: 5000,
+			id: `comment-notice-${ commentId }`,
+			isPersistent: true,
+		};
+
+		if ( 'delete' !== newStatus ) {
+			options.button = translate( 'Undo' );
+			options.onClick = () => this.setCommentStatus( commentId, previousStatus, true );
+		}
```

Comment:
@Copons I find that it helps to move bare booleans of long function signatures to an options object, so the reader can have some more context with what's happening. I would also suggest renaming the boolean from `undo` to `showNotice` since we're not attempting to compute any previous state. If we're smart about this I think we can tuck away the notice logic into other middleware.

But as a first step we might do something like this for example:
```
setCommentStatus( commentId, previousStatus, { showNotice: false } )
```

@dmsnell are you suggesting adding an undo queue, where we add an undo action to the queue every time an undoable action is fired?

I've worked with some of these in the past and I find that they undo actions tend to boil down to either computable undos ( I've liked this comment / therefore undo should unlike it ), or undos that need to reset some state, for example restoring post content or unapplying a filter to an image.

I personally would love to see something like this landing, especially for the post editor and image editor.